DBZ-3605 Add support for several Oracle data types (#4615)
debezium/debezium: documentation/modules/ROOT/pages/connectors/oracle.adoc

|`RAW`
|n/a

> |Either the raw bytes (the default), a base64-encoded String, or a base64-url-safe-encoded String, or a hex-encoded String, based on the xref:oracle-property-binary-handling-mode[`binary.handling.mode`] connector configuration property setting.

Reviewer: ```suggestion
|Depending on the setting of the xref:oracle-property-binary-handling-mode[`binary.handling.mode`] property in the connector configuration, the connector maps LOB values of this type to one of the following the semantic types:

* Raw bytes (the default)
* A base64-encoded string
* A base64-url-safe-encoded string
* A hex-encoded string.
```

Reviewer: We could apply a similar change to the description on L1398.

|===

> 
> [id="oracle-xml-types"]
> === XML type

Reviewer: ```suggestion

[id="oracle-xml-types"]
=== XML types
```

ifdef::community[]
[NOTE]
====

> Support for `XMLTYPE` is currently in incubating state, that is, the exact semantics, configuration options and so forth might change in future revisions, based on feedback we receive.

Reviewer: ```suggestion
Support for `XMLTYPE` is currently in an incubating state.
Depending on the feedback that we receive, the exact semantics, configuration options, and so forth might change in future revisions.
```

Please let us know if you encounter any problems while using these data types.
====
endif::community[]
ifdef::product[]
[IMPORTANT]
====

> Use of the `XMLTYpE` with the {prodname} Oracle connector is a Technology Preview feature only.

Reviewer: ```suggestion
Use of the `XMLTYPE` with the {prodname} Oracle connector is a Technology Preview feature only.
```

Technology Preview features are not supported with Red Hat production service level agreements (SLAs) and might not be functionally complete.
Red Hat does not recommend using them in production. These features provide early access to upcoming product features, enabling customers to test functionality and provide feedback during the development process.
For more information about the support scope of Red Hat Technology Preview features, see link:https://access.redhat.com/support/offerings/techpreview[https://access.redhat.com/support/offerings/techpreview].
====
endif::product[]
The following table describes how the connector maps XMLTYPE data types.

.Mappings for Oracle XMLTYPE data types
[cols="20%a,15%a,55%a",options="header"]
|===

> |Oracle Data Type

Reviewer: ```suggestion
|Oracle data type
```
We use sentence-style capitalization in table headings downstream. I know that this isn't consistently applied upstream, so we can either let it go here for now, or make the change and give me the task to do a thorough scrubbing of the doc set to apply the formatting.

+
NOTE: If you use the {prodname} Oracle connector with Oracle XStream, obtain the JDBC driver as part of the Oracle Instant Client package.
For more information, see xref:obtaining-oracle-jdbc-driver-and-xstreams-api-files[].

> . Download the link:https://repo1.maven.org/maven2/com/oracle/database/xml/xdb/{ojdbc8-version}/xdb-{ojdbc8-version}.jar[XDB library for Oracle] from Maven Central and extract the downloaded file to the directory that contains the {prodname} Oracle connector JAR file.

Reviewer: In your review request, you refer to changes that might be "needed for the product side", which I take to mean for use downstream. But this content is only in the Deployment topic that's conditionalized for community use. 

Do we need to include a reference to obtaining the XDB library in `/partials/modules/all-connectors/ref-deploy-oracle-kafka-connect-yaml.adoc` ([L10](https://github.com/debezium/debezium/blame/main/documentation/modules/ROOT/partials/modules/all-connectors/ref-deploy-oracle-kafka-connect-yaml.adoc#L10), and [L42](https://github.com/debezium/debezium/blame/main/documentation/modules/ROOT/partials/modules/all-connectors/ref-deploy-oracle-kafka-connect-yaml.adoc#L42))?

And probably a short topic in the Deployment overview in `oracle.adoc`, analogous to (and adjacent to)  _[Obtaining the Oracle JDBC driver_,](https://github.com/debezium/debezium/blame/main/documentation/modules/ROOT/pages/connectors/oracle.adoc#L2151) that explains why you might want to obtain the XDB library.

Author: That's correct @roldanbob.  Additionally the user will need to obtain the `xmlparserv2` dependency at 
```
<dependency>
  <groupId>com.oracle.database.xml</groupId>
  <artifactId>xmlparserv2</artifactId>
  <version>21.6.0.0</version>
</dependency>
```
I also added a blurb about the `-Djavax.xml.parsers.SAXParserFactory` argument required for XML support with Oracle, so we may need to rework how to include this elsewhere when enabling XML support.

Reviewer: Agreed. A procedure module, e.g., _Enabling support for Oracle XML data types_.

As far as adding content to the downstream deployment instructions, how shall we handle that.  Want me to open a separate PR against DBZ-3605 to make those changes? Or I could wait until you merge this and then add the downstream version of the content?
